### ISSUE TITLE
Oneof unions don't work with primitives / default values

### ISSUE THREAD

**Issue by davidvuong** [stephenh/ts-proto#102]
```proto
# test.proto
syntax = "proto3";

enum TypeA {
    x = 0;
    y = 1;
}

message Example {
    string name = 1;
    oneof test {
        TypeA my_type = 2;
        bool enabled = 3;
    }
}
```

Running this command to generate TS interfaces:

```bash
protoc --plugin=./node_modules/.bin/protoc-gen-ts_proto --ts_proto_out=. --ts_proto_opt=env=node --ts_proto_opt=oneof=unions test.proto
```

Versions:

```
"ts-proto": "^1.26.0"
libprotoc 3.11.4
```

Here is an example:

```typescript
// app.ts
import { TypeA, Example } from "./test";

const eg1 = Example.encode({
  name: "David",
  test: {
    $case: "enabled",
    enabled: true,
  },
}).finish();
console.log(Example.decode(eg1));

const eg2 = Example.encode({
  name: "David",
  test: {
    $case: "myType",
    myType: TypeA.y,
  },
}).finish();
console.log(Example.decode(eg2));

const eg3 = Example.encode({
  name: "David",
  test: {
    $case: "myType",
    myType: TypeA.x,
  },
}).finish();
console.log(Example.decode(eg3));
```

Below is the output:

```
{ name: 'David', test: { '$case': 'enabled', enabled: true } }
{ name: 'David', test: { '$case': 'myType', myType: 1 } }
{ name: 'David' }
```

Either `decode` or `encode` is omitting `test` in the last example. I expected the output to look like:

```
{ name: 'David', test: { '$case': 'myType', myType: 0 } }
```

However `test` is undefined. What am I missing?

**Comment by philikon** [comment]
Nice catch. Yes, I think that's a bug. Will provide a fix.

**Comment by philikon** [comment]
Actually, this somewhat consistent with the default (`oneof=properties`) behavior.
```ts
const eg3 = Example.encode({
  name: "David",
  myType: TypeA.x,
  enabled: undefined,
}).finish();
console.log(Example.decode(eg3));
```
also yields
```
{ name: 'David' }
```

**Comment by philikon** [comment]
The problem is definitely on the encoding side, where we'll not even emit the bytes for `myType` if its value is 0. @stephenh, what's the reason for that behavior? I definitely think we should change it for `oneof=unions`, but I'd argue it should probably be changed for `oneof=properties` as well...

**Comment by stephenh** [comment]
@philikon re "not even emit the bytes for `myType` if its value is 0", in theory that's "how protobuf is supposed to work" for default values, i.e. any primitive that has a value that is that primitive's default value, "it's a feature" to not include it on the wire, and just assume that the reader/decoder is going to treat any missing-primitive as that fields value.

Fwiw I'm not saying that's necessarily perfect, just that it's how protobuf in general works, and it does kinda make sense in a "well that is cute" sort of way. And it is technically saving bytes on the wire/on disk.

Another fwiw, I think some of this might change in the latest field presence feature of proto3, see #73 . 

I haven't fully understood the root cause of this issue yet, but just in general that is why `myType: 0` is not put in the output.

...it makes sense that would throw off our `oneof` decoding...I wonder how this is handled in other protobuf libraries. Like maybe a field being included in a `oneof` should turn off the "don't serialize it if its the default value" / i.e. "always serialize"...

**Comment by philikon** [comment]
> maybe a field being included in a `oneof` should turn off the "don't serialize it if its the default value" / i.e. "always serialize"

That's what I'm thinking.

**Comment by stephenh** [comment]
Late follow up, but @philikon 's PR is merged and released as v1.31.0.

Also, per my musing earlier in thread about "wonder what other protobuf libraries do", looks like it is actually the right thing to do to skip the "don't put default values on the wire" step for `oneof`s:

"If you set a oneof field to the default value (such as setting an int32 oneof field to 0), the "case" of that oneof field will be set, and the value will be serialized on the wire."

https://developers.google.com/protocol-buffers/docs/proto#oneof

Which makes a lot of sense.

**Comment by philikon** [comment]
Thanks!
